GameEngine/GameState.cs

```diff
@@ -1,14 +1,19 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 
```

Comment:
Does Linq provide something like a ring buffer so that we don't have to roll our own (with index + modulo arithmetic later on)?

Reply:
A quick search doesn't reveal any ring buffer in the System.Collections.Generic namespace.

Reply:
It does look like there are nuget packages that have ring buffers, but that feels like overkill to avoid either the if-statement or the %.

Reply:
After the refactor below, I'm fine with keeping it the way it is (rather than pull in a nuget package).